### PR TITLE
update verbose = 1

### LINKED ISSUE #29: optimality gap in BC algorithm
Pajarito prints an optimality gap when running the BC algorithm with ``verbose=1``. What does this mean? We've never solved a full MILP relaxation so we can't compute an optimality gap.

### LINKED ISSUE #31: need better printing between iterations
The following output is not super informative:
> Optimality Gap: 0.02247865065582623 - 0.014371448212041376 = 0.008107202443784853

Could we please have:
- description of what numbers mean (column names)
- iteration number
- abs gap
- relative gap
- print fewer digits (format nicely)

### LINKED ISSUE #33: dimension mismatch error in sparse matrix multiply
Writing my problem in JuMP (rather than convex.jl, and generating extended formulation manually). Got the following error:
> ERROR: LoadError: DimensionMismatch("")
>  in A_mul_B! at sparse/linalg.jl:47
>  in * at linalg/matmul.jl:87
>  in populatelinearmatrix at /Users/chris/.julia/v0.4/Pajarito/src/nonlinear.jl:252
>  in optimize! at /Users/chris/.julia/v0.4/Pajarito/src/nonlinear.jl:380
>  in solvenlp at /Users/chris/.julia/v0.4/JuMP/src/nlp.jl:1219
>  in solve at /Users/chris/.julia/v0.4/JuMP/src/solvers.jl:135
>  in solveGraph at /Users/chris/Dropbox/IsingModels/julia/graph_fullExt.jl:51
>  in include at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
>  in include_from_node1 at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
>  in process_options at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
>  in _start at /Applications/Julia-0.4.3.app/Contents/Resources/julia/lib/julia/sys.dylib
> while loading /Users/chris/Dropbox/IsingModels/julia/graph_fullExt.jl, in expression starting on line 79

The code was as follows (in file graph_fullExt.jl in our Dropbox, Emre):
```
pajarito_solver = PajaritoSolver(
        algorithm="OA",verbose=1,socp_disaggregator=true,time_limit=maxSecs,
        mip_solver=CplexSolver(CPX_PARAM_SCRIND=0,CPX_PARAM_EPINT=1e-8,CPX_PARAM_EPRHS=1e-8,CPX_PARAM_TILIM=60*60*10),
        cont_solver=IpoptSolver(print_level=0))

    mod = Model(solver=pajarito_solver)

    @defVar(mod, -thUB <= th[i=1:nNode, j=i+1:nNode] <= thUB)
    @defVar(mod, x[i=1:nNode, j=i+1:nNode], Bin)
    @defVar(mod, p[s=1:nSamp])
    @defVar(mod, t[s=1:nSamp])

    @setNLObjective(mod, Min, 1.0/nSamp * sum{t[s], s=1:nSamp}) #optionally use 1.0/nSamp multiplier

    @addNLConstraint(mod, tGE[s=1:nSamp], t[s] >= exp(p[s]))

    @addConstraint(mod, pEq[s=1:nSamp], p[s] == -sum{samps[s,i] * samps[s,j] * th[i,j], i=1:nNode, j=i+1:nNode})

    @addConstraint(mod, deg[k=1:nNode], sum{x[k,j], j=k+1:nNode} + sum{x[i,k], i=1:k-1} <= maxDeg)

    #big-M constraints (is it possible to replace with indicator constraints? pajarito compatibility?)
    @addConstraint(mod, thU[i=1:nNode, j=i+1:nNode], th[i,j] <= thUB * x[i,j])
    @addConstraint(mod, thL[i=1:nNode, j=i+1:nNode], th[i,j] >= -thUB * x[i,j])

    stat = solve(mod)
```

The error only started when I took the exponential cones out of the objective by adding the t variables and the tGE constraints.

### PR DESCRIPTION
**Pull request by emreyamangil**
This PR updates the print level when verbose flag is set to 1.

Closes #33
Closes #29 
Closes #31

### CODE REVIEW

File: src/conic.jl [hunk 1]
```diff
@@ -772,23 +772,29 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
...
     else
-        (m.verbose > 0) && println("Conic Relaxation Infeasible")
+        (m.verbose > 1) && println("Conic Relaxation Infeasible")
         m.status = :Infeasible
         return       
     end
 
+    (m.verbose > 0) && println("\nPajarito started...\n")
+    (m.verbose > 0) && println("MICONE algorithm $(m.algorithm) is chosen.")
+    (m.verbose > 0) && println("MICONE has $(m.numVar) variables, $(m.numConstr) constraints, $(m.numSpecCones) nonlinear cones.")
```
Comment (mlubin) [inline src/conic.jl]: linear constraints?

File: src/conic.jl [hunk 2]
```diff
@@ -929,7 +935,7 @@ function MathProgBase.optimize!(m::PajaritoConicModel)
 
         # add supporting hyperplanes
         optimality_gap = m.objval - mip_objval 
-        (m.verbose > 0) && println("Optimality Gap: $(m.objval) - $(mip_objval) = $(optimality_gap)")
+        (m.verbose > 0) && (m.algorithm == "OA") && @printf "%9d   %13.2f   %15.2f   %14.2f   %13.2f\n" iter mip_objval dcp_objval optimality_gap m.objval
```
Comment (mlubin) [inline src/conic.jl]: Do you print a header describing these columns?

Comment (emreyamangil) [inline src/conic.jl]: yupp!! right below at the initial algorithm call for OA

File: src/nonlinear.jl [hunk 3]
```diff
@@ -248,7 +248,7 @@ function populatelinearmatrix(m::PajaritoModel)
         push!(A_I,row); push!(A_J, jac_J[k]); push!(A_V, jac_V[k])
     end
 
-    m.A = sparse(A_I, A_J, A_V)
+    m.A = sparse(A_I, A_J, A_V, m.numConstr-m.numNLConstr, m.numVar)
```
Comment (mlubin) [inline src/nonlinear.jl]: Can you add a test that triggers this error so it doesn't repeat in the future?

Comment (emreyamangil) [inline src/nonlinear.jl]: Will do!!

**Comment by mlubin** [comment]
Travis is passing, I'll merge this and future tweaks to the printing can go into a new PR

**Pull request merged**